### PR TITLE
added typography component

### PR DESCRIPTION
**Pull request by ufoBeliever**
added typography component

closing issue https://github.com/igorizviekov/mt-booking_app/issues/15

### CODE REVIEW

**Review by igorizviekov — COMMENTED** [review]
No fix is needed here, but just curious if we need to swap refactor and feat of these 2 commits.
image

File: tailwind.config.js [hunk 1]
```diff
@@ -2,7 +2,14 @@
 module.exports = {
   content: ['./src/**/*.{js,jsx,ts,tsx}'],
   theme: {
-    extend: {},
+    extend: {
+      colors: {
+        'primary-color': 'var(--primary-color)',
```
Comment (igorizviekov) [inline tailwind.config.js:9]: Why not specify colors here, in the config file?

File: tailwind.config.js [hunk 2]
```diff
@@ -11,5 +13,13 @@ module.exports = {
       },
     },
   },
-  plugins: [],
+  plugins: [
+    plugin(function ({ addUtilities, addComponents, e, config }) {
+      // Add your custom styles here
```
Comment (igorizviekov) [inline tailwind.config.js:18]: I don't see you use any plugins. Can you add one?

File: src/sections/Listings/LIstings.tsx [hunk 3]
```diff
@@ -40,7 +40,7 @@ export const Listings: React.FC<ListingsProps> = ({ title }) => {
     );
   }
   return (
-    <div className='p-4'>
+    <div className='p-4 dark:bg-neutral-500'>
```
Comment (igorizviekov) [inline src/sections/Listings/LIstings.tsx]: Where do you get theme context from?

File: src/App.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import Listings from './sections/Listings';
 import 'antd/dist/reset.css';
-import './styles/index.css';
+import './styles/index.scss';
```
Comment (igorizviekov) [inline src/App.tsx:4]: Can you [BEM](https://css-tricks.com/bem-101/) in a scss file to apply styles

File: src/sections/Listings/LIstings.tsx [hunk 5]
```diff
@@ -40,10 +40,12 @@ export const Listings: React.FC<ListingsProps> = ({ title }) => {
     );
   }
   return (
-    <div className='p-4 dark:bg-neutral-500'>
-      <h2 className='mb-4 text-4xl text-text-color-1'>{title}</h2>
+    <div className='p-4'>
+      <h2 className='mb-4 text-4xl text-text-color-1 lg:tracking-widest dark:text-white'>
```
Comment (igorizviekov) [inline src/sections/Listings/LIstings.tsx]: can you apply default styles for sm and them `md` `lg` and `xl`

File: src/components/Typography/Typography.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+
+interface TypographyProps {
+  size: number;
+  label: string;
+}
```
Comment (igorizviekov) [inline src/components/Typography/Typography.tsx:6]: 1.  can you move all types to a separate file f.e. typography.types.d.ts
2. can yo add [jsdoc](https://jsdoc.app/about-getting-started.html) comments for all properties
3. can you rename interface to ITypographyProps and follow this naming convention for interface names

File: tailwind.config.js (hunk 2 above)
Comment (ufoBeliever) [reply tailwind.config.js:18]: https://github.com/igorizviekov/mt-booking_app/commit/0dd59e9b56006937e1ee0f6441b2373747107da0

File: src/sections/Listings/LIstings.tsx (hunk 3 above)
Comment (ufoBeliever) [reply src/sections/Listings/LIstings.tsx]: https://github.com/igorizviekov/mt-booking_app/commit/934af7151ebc2a241038463b34aa6250119bc21f

File: src/App.tsx (hunk 4 above)
Comment (ufoBeliever) [reply src/App.tsx:4]: https://github.com/igorizviekov/mt-booking_app/commit/a8c28341f2fd96cd75a42ba8a1e8e714e4184007

File: src/sections/Listings/LIstings.tsx (hunk 5 above)
Comment (ufoBeliever) [reply src/sections/Listings/LIstings.tsx]: https://github.com/igorizviekov/mt-booking_app/commit/d711e0e558074667d2f7f521670bfe51c17dca26

File: src/components/Typography/Typography.tsx (hunk 6 above)
Comment (ufoBeliever) [reply src/components/Typography/Typography.tsx:6]: https://github.com/igorizviekov/mt-booking_app/commit/9890b68f16912f4f8fe1b42285ed492a6e78b23c
https://github.com/igorizviekov/mt-booking_app/commit/b0640ecb59e66c95da4a773d7359b4cdb0096d71

File: src/App.tsx (hunk 4 above)
Comment (igorizviekov) [reply src/App.tsx:4]: update scss file with refactored nested class names to be` &__{{ className }}`

**Comment by igorizviekov** [comment]
closing as outdated

**Comment by igorizviekov** [comment]
closing as outdated

**Pull request closed without merging**